internal/bootstrap/deployer.go

```diff
@@ -460,7 +460,7 @@ func (b *baseDeployer) AddControllerApplication(ctx context.Context, info Deploy
 				Since:  ptr(b.clock.Now()),
 			},
 		},
-		applicationservice.AddUnitArg{UnitName: unitName},
+		applicationservice.AddUnitArg{},
```

Comment:
This is suspect. The `unitName` variable is created above and returned for use by the caller to `AddControllerApplication`, later used to call the `CompleteProcess` method.  It is a side effect of bootstrap that is happens to be okay, not a direct action.

At the very least, a comment is needed for the code block at line 442.

Reply:
Yes, you're right. Long term, CreateApplication should probably return the unit names/ids.

But I think for now, a comment is sufficient. I will add